pkg/config/apiserver.go

```diff
@@ -114,6 +114,52 @@ func (a *API) register(c *restful.Container) {
 		Reads(&pb.ServiceConfig{}).
 		Writes(APIResponse{}))
 
+	// Adapters
+	ws.Route(ws.
+		GET("/scopes/{scope}/adapters").
+		To(a.getAdaptersOrDescriptors).
+		Doc("Gets named adapter configurations.").
+		Param(ws.PathParameter("scope", "scope").DataType("string")).
+		Writes(APIResponse{}))
+
+	ws.Route(ws.
+		PUT("/scopes/{scope}/adapters").
+		To(a.putAdaptersOrDescriptors).
+		Doc("Creates or replaces named adapter configurations.").
+		Param(ws.PathParameter("scope", "scope").DataType("string")).
+		Reads(&pb.GlobalConfig{}).
+		Writes(APIResponse{}))
+
+	ws.Route(ws.
+		DELETE("/scopes/{scope}/adapters").
+		To(a.deleteAdaptersOrDescriptors).
+		Doc("Deletes named adapter configurations.").
+		Param(ws.PathParameter("scope", "scope").DataType("string")).
+		Writes(APIResponse{}))
+
+	// Descriptors
+	ws.Route(ws.
```

Comment:
There are `PATCH` methods to come later (they are in the spec), that let you update individual descriptors and adapters.